metadata-builders/src/main/java/com/linkedin/metadata/builders/search/MLModelIndexBuilder.java

```diff
@@ -0,0 +1,203 @@
+package com.linkedin.metadata.builders.search;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+import javax.annotation.Nonnull;
+
+import com.linkedin.common.DatasetUrnArray;
+import com.linkedin.common.InstitutionalMemory;
+import com.linkedin.common.Ownership;
+import com.linkedin.common.Status;
+import com.linkedin.common.urn.MLModelUrn;
+import com.linkedin.data.template.RecordTemplate;
+import com.linkedin.data.template.StringArray;
+import com.linkedin.metadata.search.MLModelDocument;
+import com.linkedin.metadata.snapshot.MLModelSnapshot;
+import com.linkedin.ml.metadata.BaseData;
+import com.linkedin.ml.metadata.CaveatsAndRecommendations;
+import com.linkedin.ml.metadata.EthicalConsiderations;
+import com.linkedin.ml.metadata.EvaluationData;
+import com.linkedin.ml.metadata.IntendedUse;
+import com.linkedin.ml.metadata.MLModelFactorPrompts;
+import com.linkedin.ml.metadata.MLModelProperties;
+import com.linkedin.ml.metadata.Metrics;
+import com.linkedin.ml.metadata.QuantitativeAnalyses;
+import com.linkedin.ml.metadata.TrainingData;
+
+
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class MLModelIndexBuilder extends BaseIndexBuilder<MLModelDocument> {
+
+    public MLModelIndexBuilder() {
+        super(Collections.singletonList(MLModelSnapshot.class), MLModelDocument.class);
+    }
+
+    @Nonnull
+    private static String buildBrowsePath(@Nonnull MLModelUrn urn) {
+        return ("/" + urn.getOriginEntity() + "/"  + urn.getPlatformEntity().getPlatformNameEntity() + "/" + urn.getMlModelNameEntity())
+            .replace('.', '/').toLowerCase();
+    }
+
+    /**
+     * Given model urn, this returns a {@link MLModelDocument} model that has urn, dataset name, platform and origin fields set
+     *
+     * @param urn {@link MLModelUrn} that needs to be set
+     * @return {@link MLModelDocument} model with relevant fields set that are extracted from the urn
+     */
+    @Nonnull
+    private static MLModelDocument setUrnDerivedFields(@Nonnull MLModelUrn urn) {
+        return new MLModelDocument()
+            .setName(urn.getMlModelNameEntity())
+            .setOrigin(urn.getOriginEntity())
+            .setPlatform(urn.getPlatformEntity().getPlatformNameEntity())
+            .setUrn(urn)
+            .setBrowsePaths(new StringArray(Collections.singletonList(buildBrowsePath(urn))));
```

Comment:
Are you planning to support browse feature on ML Models? `browsePaths` will make sense in that case. I didn't see  browse method in the resource nor the client. It will be great if you can clarify the same.

Reply:
I just set it since the field is in every BaseDocument extension. I can take it out, looks like other non-browseable indices just leave it null.